app/src/main/java/com/garethevans/church/opensongtablet/StageMode.java

```diff
@@ -2599,6 +2772,91 @@ private void customButtonAction(String s) {
                 }
                 break;
         }
+
+        pageButtonAlpha("");
+
+        // Restore full screen gesture activity after a delay
+        Handler h = new Handler();
+        h.postDelayed(() -> StaticVariables.ignoreGesture = false, 300);
+    }
+
+    // IV - Long click activites added
+    private void customButtonLongPressAction(String s) {
+        // Flag to ignore full screen long press activity for a short while
+        StaticVariables.ignoreGesture = true;
+
+        switch (s) {
+            case "":
+            default:
+                FullscreenActivity.whattodo = "quicklaunch";
+                openFragment();
+                break;
+
+            case "editsong":
+            case "editsongpdf":
+            case "changetheme":
+            case "autoscale":
+            case "changefonts":
+            case "profiles":
+            case "gestures":
+            case "footpedal":
+            case "transpose":
+            case "fullsearch":
+            case "randomsong":
+            case "abcnotation_edit":
+            case "abcnotation":
+            case "abcnotation_editsong":
+            case "showmidicommands":
+            case "showlyrics":
+            case "inc_autoscroll_speed":
+            case "dec_autoscroll_speed":
+            case "toggle_autoscroll_pause":
+                if (s.equals("editsong") && !justSong(StageMode.this) && !FullscreenActivity.isPDF) {
+                    ShowToast.showToast(StageMode.this);
+                } else {
+                    if (FullscreenActivity.isPDF && s.equals("editsong")) {
+                        s = "extractPDF";
+                    }
+                    FullscreenActivity.whattodo = s;
+                    openFragment();
+                }
+                break;
+
+            case "showchords":
+                // Display the native chords
+                StaticVariables.showCapoInChordsFragment = false;
+                FullscreenActivity.whattodo = "page_chords";
+                openFragment();
+                break;
```

Comment:
This explains the PopUpCustomChordsFragment behaviour.  Makes sense now.

Reply:
OK